src/deadline/client/cli/_groups/manifest_group.py

```diff
@@ -0,0 +1,206 @@
+import click
+
+from deadline.job_attachments.api.manifest import (
+    _manifest_snapshot,
+)
+
+from ...exceptions import NonValidInputError
+from .._common import _handle_error
+from .click_logger import ClickLogger
+
+
+@click.group(name="manifest")
+@_handle_error
+def cli_manifest():
+    """
+    Commands to work with AWS Deadline Cloud Job Attachments.
+    """
+
+
+@cli_manifest.command(
+    name="snapshot",
+    help="BETA - Generates a snapshot of files in a directory root as a Job Attachment Manifest.",
+)
+@click.option("--root", required=True, help="The root directory to snapshot. ")
+@click.option(
+    "-d",
+    "--destination",
+    default=None,
+    help="Destination directory where manifest is created. Defaults to the manifest root directory.",
+)
+@click.option(
+    "-n",
+    "--name",
+    default=None,
+    help="Name of the manifest. A timestamp is added YYYY-MM-DD-HH-MM-SS for versioning.",
+)
+@click.option(
+    "-i",
+    "--include",
+    default=None,
+    help="Glob syntax of files and directories to include in the manifest. Can be provided multiple times.",
+)
+@click.option(
+    "-e",
+    "--exclude",
+    default=None,
+    help="Glob syntax of files and directories to exclude in the manifest. Can be provided multiple times.",
+    multiple=True,
+)
+@click.option(
+    "-ie",
+    "--include-exclude-config",
+    default=None,
+    help="Include and exclude config of files and directories to include and exclude. Can be a json file or json string.",
+    multiple=True,
+)
+@click.option("--diff", default=None, help="File Path to Asset Manifest to diff against.")
+@click.option("--json", default=None, is_flag=True, help="Output is printed as JSON for scripting.")
+@_handle_error
+def manifest_snapshot(
+    root: str,
+    destination: str,
+    name: str,
+    include: List[str],
+    exclude: List[str],
+    include_exclude_config: str,
+    diff: str,
+    json: bool,
+    **args,
+):
+    """
+    Creates manifest of files specified by root directory.
+    """
+    logger: ClickLogger = ClickLogger(is_json=json)
+    if not os.path.isdir(root):
+        raise NonValidInputError(f"Specified root directory {root} does not exist.")
+
+    if destination and not os.path.isdir(destination):
+        raise NonValidInputError(f"Specified destination directory {destination} does not exist.")
+    elif destination is None:
+        destination = root
+        logger.echo(f"Manifest creation path defaulted to {root} \n")
+
+    manifest_out = _manifest_snapshot(
+        root=root,
+        destination=destination,
+        name=name,
+        include=include,
+        exclude=exclude,
+        include_exclude_config=include_exclude_config,
+        diff=diff,
+        logger=logger,
+    )
+    if manifest_out:
+        logger.json(dataclasses.asdict(manifest_out))
+
+
+@cli_manifest.command(
+    name="diff", help="BETA - Compute a directory root diff of new, modified or deleted files."
```

Comment:
nit: I think the Help text could be made clearer for the customer with a mention of the manifest against which the diff is being taken? Thoughts?

Reply:
Agreed! How about "Compute the file difference of a root directory against an existing manifest for new, modified or deleted files."